src/main/java/randoop/generation/ObjectPool.java

```diff
@@ -0,0 +1,206 @@
+
+      Object generatedObjectValue = null;
+      ExecutionOutcome lastOutcome = eseq.getResult(eseq.sequence.size() - 1);
+      if (lastOutcome instanceof NormalExecution) {
+        generatedObjectValue = ((NormalExecution) lastOutcome).getRuntimeValue();
+      }
+
+      if (generatedObjectValue != null) {
+        this.addOrUpdate(generatedObjectValue, genSeq);
+      }
+    }
+  }
+
+  /**
+   * Check if the object pool is empty.
+   *
+   * @return true if the pool is empty, false otherwise
+   */
+  public boolean isEmpty() {
+    return this.objPool.isEmpty();
+  }
+
+  /**
+   * Get the size of the object pool.
+   *
+   * @return the number of objects in the pool
+   */
+  public int size() {
+    return this.objPool.size();
+  }
+
+  /**
+   * Add a new object and its associated sequences to the pool.
+   *
+   * @param obj the object to be added
+   * @param sequences the sequences associated with the object
+   */
+  public void put(Object obj, SimpleList<Sequence> sequences) {
+    this.objPool.put(obj, sequences);
+  }
+
+  /**
+   * Get the sequences associated with a specific object.
+   *
+   * @param obj the object
+   * @return the sequences associated with the object
+   */
+  public SimpleList<Sequence> get(Object obj) {
+    return this.objPool.get(obj);
+  }
+
+  /**
+   * Get a list of all objects in the pool.
+   *
+   * @return a list of all objects
+   */
+  public List<Object> getObjects() {
+    // return List.copyOf(this.objPool.keySet());
+    return new ArrayList<>(this.objPool.keySet());
+  }
+
+  /**
+   * Add a new sequence to an object's associated sequences or create a new entry if the object is
+   * not in the pool.
+   *
+   * @param obj the object
+   * @param seq the sequence to be added
+   */
+  @SuppressWarnings("unchecked")
+  /*
+  // Java 9 version
+  public void addOrUpdate(Object obj, Sequence seq) {
+    if (this.objPool.containsKey(obj)) {
+      SimpleList<Sequence> existingSequences = this.objPool.get(obj);
+      this.objPool.put(
+          obj, new ListOfLists<>(existingSequences, new SimpleArrayList<>(List.of(seq))));
+    } else {
+      this.objPool.put(obj, new SimpleArrayList<>(List.of(seq)));
+    }
+  }
+   */
+  public void addOrUpdate(Object obj, Sequence seq) {
+    if (this.objPool.containsKey(obj)) {
+      SimpleList<Sequence> existingSequences = this.objPool.get(obj);
+
+      // Create a new SimpleArrayList and add 'seq' to it
+      SimpleArrayList<Sequence> newList = new SimpleArrayList<>();
+      newList.add(seq);
+
+      // Use ListOfLists with existingSequences and newList
+      this.objPool.put(obj, new ListOfLists<>(existingSequences, newList));
+    } else {
+      // Create a new SimpleArrayList and add 'seq' to it for the else case
+      SimpleArrayList<Sequence> newList = new SimpleArrayList<>();
+      newList.add(seq);
+
+      this.objPool.put(obj, newList);
+    }
+  }
+
+  /**
+   * Filter the sequences in the pool by their type.
```

Comment:
"Filter" is ambiguous: it can mean both "remove" and "retain".  Please clarify.

Reply:
Changed to:
  /**
   * Get a list of sequences that create objects of a specific type.
   *
   * @param t the type of objects that the sequences create
   * @return a list of sequences that create objects of the specified type
   */